docs/analysis/datadog.md

```diff
@@ -41,6 +41,48 @@ stringData:
 
 `apiVersion` here is different from the `apiVersion` from the Datadog configuration above.
 
+!!! important
+    ###### Namespaced secret
+    Datadog integration supports referring to secrets inside the same namespace as argo-rollouts (by default)
+    or referring to a secret in the same namespace as the `AnalysisTemplate`.
+
+    To use a secret from the `AnalysisTemplate` namespace, include a `secretRef` section in the template, specifying the `name` of the secret and setting the `namespaced` property to `true`.
+
+    The process for retrieving Datadog credentials is as follows:
+    1. **If a `secretRef` is defined in the `AnalysisTemplate`:** Argo Rollouts will search for the secret with the specified name in the namespace where the template resides.
+    2. **If the secret is not found in the specified namespace:** Argo Rollouts will then check the environment variables.
+    3. **If the credentials are not found in environment variables:** Argo Rollouts will look for a secret named "Datadog" in the namespace where Argo Rollouts itself is deployed.
+
+--- 
+
+Let me know if there's anything else you'd like to adjust!
+
+    ```yaml
+    apiVersion: argoproj.io/v1alpha1
+    kind: AnalysisTemplate
+    metadata:
+      name: loq-error-rate
+    spec:
+      args:
+      - name: service-name
+      metrics:
+      - name: error-rate
+        interval: 5m
+        successCondition: result <= 0.01
+        failureLimit: 3
+        provider:
+          datadog:
+            apiVersion: v2
+            interval: 5m
+            secretRef:
+              name: "mysecret"
+              namespaced: true
```

Comment:
I knid of think we just drop this field `namespaced`, i think the presence of a secretRef just means look up secret in the namespace of the AnalysisTemplate. At a later date we can then add a `namespace`  field to secretRef if we want to support cross namespace lookups. Thoughts?

Reply:
Hi @zachaller 
Thanks for the review and the input.

When implementing this feature we discussed the different use-cases and we thought about the following ones:

1. As an argo rollout owner, you also own the integration with a single DataDog account.
2. As an argo rollout owner, you also own the integration with several DataDog accounts
3. As an argo rollout owner, you only own argo rollout and your users own the DataDog ones.

The first case is the currently supported case.

The second case corresponds to `namespaced: false`, and let the "rollouts administrator" to manage different DataDog integrations by providing well-known secret names to the namespace users.

The third one corresponds to `namespaced: true`, where namespace users were completely autonomous to manage the DataDog account integration.

We did consciously exclude the option to have a configurable `namespace` string to maximise security.
Indeed, we want the owners of, say, `namespace-a` `AnalysisTemplate`s and DataDog integrations to have confidence that they will control who can access their metrics.

Having a configurable free text `namespace` would allow users of any other namespace to indirectly access DataDog metrics, which would then break the namespace principles.

WDYT?